cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/SlangCompilerImpl.java

```diff
@@ -66,4 +68,43 @@ public Executable preCompile(SlangSource source) {
 
         return executable;
     }
+
+    @Override
+    public Set<SystemProperty> loadSystemProperties(SlangSource source) {
+        try {
+            ParsedSlang parsedSlang = parseSource(source);
+            return extractProperties(parsedSlang);
+        } catch (Throwable ex) {
+            throw new RuntimeException(
+                    "Error loading properties file: " + source.getName() + " nested exception is " + ex.getMessage(),
+                    ex
+            );
+        }
+    }
+
+    private ParsedSlang parseSource(SlangSource source) {
+        ParsedSlang parsedSlang = yamlParser.parse(source);
+        if (!ParsedSlang.Type.SYSTEM_PROPERTY_FILE.equals(parsedSlang.getType())) {
+            throw new RuntimeException("Source: " + parsedSlang.getName() + " is not a valid system property file.");
+        }
+        return parsedSlang;
+    }
+
+    private Set<SystemProperty> extractProperties(ParsedSlang parsedSlang) {
+        String namespace = parsedSlang.getNamespace();
+        Map<String, Object> rawSystemProperties = parsedSlang.getProperties();
+        Set<SystemProperty> properties = new HashSet<>();
+        Set<Map.Entry<String, Object>> entrySet = rawSystemProperties.entrySet();
```

Comment:
Did you check if you call this method with a parsedSlang object that is not system properties that there are no NullPointer exceptions?